keps/sig-node/4569-cgroup-v1-deprecation/README.md

```diff
@@ -0,0 +1,685 @@
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+<!--
+**ACTION REQUIRED:** In order to merge code into a release, there must be an
+issue in [kubernetes/enhancements] referencing this KEP and targeting a release
+milestone **before the [Enhancement Freeze](https://git.k8s.io/sig-release/releases)
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+- [ ] (R) Graduation criteria is in place
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Deprecate the support for cgroup v1 in Kubernetes, aligning with the industry's move towards cgroup v2 as the default for Linux kernel resource management and isolation.
+
+## Motivation
+
+cgroup v2 offers significant improvements over cgroup v1 in terms of feature set, uniformity of interface, and scalability. With major Linux distributions and container runtimes supporting cgroup v2, it is timely for Kubernetes to deprecate cgroup v1 support to streamline maintenance and encourage the adoption of newer, more secure, and more performant technologies.
+
+
+### Goals
+
+- Announce the deprecation of cgroup v1 support in Kubernetes.
+- Provide clear migration paths for users running Kubernetes on systems using cgroup v1.
+- Ensure test coverage
```

Comment:
nit: I think we should drop this from a goal.

Reply:
I wanted to highlight that before we move to beta and enable that feature gate by default, we have to make sure that all jobs are migrated to cgroup v2. maybe there is a better way to say that.

Reply:
If we migrate all jobs to cgroup v2 while this KEP is just beta, wouldn't we lose all test coverage for v1? We should still have both test coverage until we drop the support.

Reply:
```suggestion
- Ensure all features coverage by running all tests on cgroup v2 (while some may still run on cgroup v1 to test back compatibility)
```